File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    // Prompt user for string of text
    string s = get_string("Text: ");
    int letter_count = 0;
    int word_count = 1;
    int sentence_count = 0;

    // Identify and count parts of the string via else if statements
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letter_count++;
        }
        else if (s[i] == ' ')
        {
            word_count++;
        }
        else if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            sentence_count++;
        }
    }


    // Apply Coleman-Liau index
    float grade = 0.0588 * (100 * (float) letter_count / (float) word_count) - 0.296 * (100 * (float) sentence_count / 
                  (float) word_count) - 15.8;
    if (grade < 16 && grade > 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: readability.py
from cs50 import get_string

# Prompt user for string of text
s = get_string("Text:")

letter_count = 0
word_count = 1
sentence_count = 0

# Use else if statements to identify and count string components

for c in s:
    if str.isalpha(c):
        letter_count += 1
    elif c is ' ':
        word_count += 1
    elif c is '.' or c is '?' or c is '!':
        sentence_count += 1

g = round(0.0588 * (100 * letter_count) / word_count - 0.296 * (100 * sentence_count) / word_count - 15.8)

if g > 0 and g < 16:
    print(f"Grade: {g}")
elif g >= 16:
    print("Grade: 16+")
else:
    print("Before Grade 1")